From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Tue, 21 Jul 2020 05:47:29 +1200
Subject: Allow steps in docker build script to be skipped using environment
 variables

Skip default-judgehost-chroot image by default because it's only
needed for dj_make_chroot_docker.

Skip help message by default because it is no longer accurate (some of
the images might not exist, and the instructions for pushing to Docker
Hub aren't relevant).
---
 docker/build.sh | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/docker/build.sh b/docker/build.sh
index 94dd8f1d5..235725284 100755
--- a/docker/build.sh
+++ b/docker/build.sh
@@ -8,7 +8,7 @@ then
 	exit 1
 fi
 
-if false; then
+if [ "${BUILD_DOWNLOAD_RELEASE-0}" != 0 ]; then
 URL=https://www.domjudge.org/releases/domjudge-${VERSION}.tar.gz
 FILE=domjudge.tar.gz
 
@@ -23,17 +23,22 @@ fi
 echo "[ok] DOMjudge version ${VERSION} downloaded as domjudge.tar.gz"; echo
 fi
 
+if [ "${BUILD_FROM_SOURCE-1}" != 0 ]; then
 echo "[..] Boostrapping..."
 tar c -C .. --exclude=./.git --exclude="./docker/*.tar.gz" . |
 	docker build --target=dist -t domjudge/dist:${VERSION} -f docker/domserver/Dockerfile.source -
 docker run --rm domjudge/dist:${VERSION} tar cz -C /domjudge-src domjudge > domjudge.tar.gz
 docker rmi --no-prune domjudge/dist:${VERSION}
 echo "[ok] Done boostrapping"
+fi
 
+if [ "${BUILD_DOMSERVER-1}" != 0 ]; then
 echo "[..] Building Docker image for domserver using intermediate build image..."
 docker build -t domjudge/domserver:${VERSION} -f domserver/Dockerfile .
 echo "[ok] Done building Docker image for domserver"
+fi
 
+if [ "${BUILD_JUDGEHOST-1}" != 0 ]; then
 echo "[..] Building Docker image for judgehost using intermediate build image..."
 docker build -t domjudge/judgehost:${VERSION}-build -f judgehost/Dockerfile.build .
 docker rm -f domjudge-judgehost-${VERSION}-build > /dev/null 2>&1 || true
@@ -44,11 +49,15 @@ docker rm -f domjudge-judgehost-${VERSION}-build
 docker rmi domjudge/judgehost:${VERSION}-build
 docker build -t domjudge/judgehost:${VERSION} -f judgehost/Dockerfile .
 echo "[ok] Done building Docker image for judgehost"
+fi
 
+if [ "${BUILD_DEFAULT_JUDGEHOST_CHROOT-0}" != 0 ]; then
 echo "[..] Building Docker image for judgehost chroot..."
 docker build -t domjudge/default-judgehost-chroot:${VERSION} -f judgehost/Dockerfile.chroot .
 echo "[ok] Done building Docker image for judgehost chroot"
+fi
 
+if [ "${BUILD_PRINT_HELP-0}" != 0 ]; then
 echo "All done. Image domjudge/domserver:${VERSION} and domjudge/judgehost:${VERSION} created"
 echo "If you are a DOMjudge maintainer with access to the domjudge organization on Docker Hub, you can now run the following command to push them to Docker Hub:"
 echo "$ docker push domjudge/domserver:${VERSION} && docker push domjudge/judgehost:${VERSION} && docker push domjudge/default-judgehost-chroot:${VERSION}"
@@ -57,3 +66,4 @@ echo "$ docker tag domjudge/domserver:${VERSION} domjudge/domserver:latest && \
 docker tag domjudge/judgehost:${VERSION} domjudge/judgehost:latest && \
 docker tag domjudge/default-judgehost-chroot:${VERSION} domjudge/default-judgehost-chroot:latest && \
 docker push domjudge/domserver:latest && docker push domjudge/judgehost:latest && docker push domjudge/default-judgehost-chroot:latest"
+fi
-- 
2.20.1

